TextFormat.md

```diff
@@ -1,64 +1,81 @@
-assembly languages have).
-
-There is no requirement to use JavaScript syntax; this format is not intended to
-be evaluated or translated directly into JavaScript. There may also be
-substantive reasons to use notation that is different than JavaScript (for
-example, WebAssembly has a 32-bit integer type, and it should be represented
-in the text format, since that is the natural thing to do for WebAssembly,
-regardless of JavaScript not having such a type). On the other hand,
-when there are no substantive reasons and the options are basically
-bikeshedding, then it does make sense for the text format to match existing
-conventions on the Web (for example, curly braces, as in JavaScript and CSS).
-
-The text format isn't uniquely representable. Multiple textual files can assemble
-to the same binary file, for example whitespace isn't relevant and memory initialization
-can be broken out into smaller pieces in the text format.
-
-The text format is precise in that values that cannot be accurately represented in the
-binary format are considered invalid text. Floating-point numbers are therefore
-represented as hexadecimal floating-point as specified by the C99 standard, which
-IEEE-754-2008 section 5.12.3 also specifies. The textual format may be improved to also
-support more human-readable representations, but never at the cost of accurate representation.
-
-# Official Text Format
-
-WebAssembly currently doesn't have a final, official, text format. As detailed above the
-main purpose of the text format will be for human consumption, feedback from humans on
-readability will therefore factor into standardizing a text format.
-
-There are, however, prototype syntaxes which are used to bring up WebAssembly: it's easier
-to develop using a text format than it is with a binary format, even if the ultimate
-WebAssembly format will be binary. Most of these prototypes use [s-expressions][] because they
-can easily represent expression trees and [ASTs](Semantics.md) (as opposed to CFGs)
-and don't have much of a syntax to speak of (avoiding syntax bikeshed discussions).
+WebAssembly does not yet have a standardized text format that encodes function
+bodies in addition to module structure, data segments and other program
+metadata in a way that is eqivalent to the [binary format](BinaryEncoding.md).
+WebAssembly does, however, have a specified
+[textual representation](#linear-bytecode) of function bodies which should be displayed in browsers and other tools when [debugging](#debug-symbol-integration)
+modules.
+
+# Linear bytecode
+
+WebAssembly function bodies encode bytecode instructions which have a
+specified textual representation. A linear presentation of these sequences of
+instructions is the only textual encoding of function bodies which can
+accurately represent the computations of the
+[structured stack machine](Semantics.md). Therefore, these linear instructions
+are a minimum necessary part of any text format.
```

Comment:
I don't think the statement is technical true either. A structured expression based text format can also encode the function code accurately, so the linear presentation is not the 'only textual encoding of function bodies which can accurately represent the computations'. If you have an example to demonstrate otherwise then please let us know? 

Expressions can be very useful to readers of the code as they present the data flow and live range for many common high frequency code patterns, so I am confident that there will be tools that use expression based patterns and other date flow patterns to assist developers.

Wasm is designed for quick decoding to validated SSA form, so this SSA form could also be considered a canonical form. There could be multiple ways to encode the same SSA form in the binary format, with the same basic sequence of instructions but with variation in use of local variables versus `pick` operators etc to connect operators. This SSA form can be presented in a range of text formats, including a functional style which is a long way from a linear format. I also think it is likely that some tools will decode to SSA form as part of the text format presentation canonicalizing away some of the encoding details, then when `pick` is added this can simply be an encoding efficiency option.